odoo/orm/models.py

```diff
@@ -197,31 +197,53 @@ class MetaModel(api.Meta):
         Its main purpose is to register the models per module.
     """
     module_to_models = defaultdict(list)
+    models = set()
 
     def __new__(meta, name, bases, attrs):
         # this prevents assignment of non-fields on recordsets
         attrs.setdefault('__slots__', ())
         # this collects the fields defined on the class (via Field.__set_name__())
         attrs.setdefault('_field_definitions', [])
 
-        if attrs.get('_register', True):
-            # determine '_module'
-            if '_module' not in attrs:
-                module = attrs['__module__']
-                assert module.startswith('odoo.addons.'), \
-                    f"Invalid import of {module}.{name}, it should start with 'odoo.addons'."
-                attrs['_module'] = module.split('.')[2]
+        if any(not isinstance(base, MetaModel) for base in bases):
+            raise TypeError('The Odoo models should only contain Odoo model without any other python classes.')
 
-            _inherit = attrs.get('_inherit')
-            if _inherit and isinstance(_inherit, str):
-                # TODO: add an exception: TypeError(f"'_inherit' property of model {name!r} should be a list: {_inherit!r}.")
-                attrs.setdefault('_name', _inherit)
-                attrs['_inherit'] = [_inherit]
-
-            if not attrs.get('_name'):
-                attrs['_name'] = class_name_to_model_name(name)
-
-        return super().__new__(meta, name, bases, attrs)
+        if not attrs.get('_register', True):
+            return super().__new__(meta, name, bases, attrs)
+
+        inherit_models = list({base._name for base in bases if base._name})
```

Comment:
I think you should use the set for this which will disallow adding multiple times the same model name and keep the order.
```suggestion
        inherit_models = OrderedSet(base._name for base in bases if base._name)
```